docs/en/operations/settings/settings.md

```diff
@@ -4067,6 +4067,18 @@ Result:
 └─────┴─────┴───────┘
 ```
 
+## splitby_max_substring_behavior {#splitby-max-substring-behavior}
+
+Controls how functions [splitBy*()](../../sql-reference/functions/splitting-merging-functions.md) with given `max_substring` argument behave.
```

Comment:
Honestly this seems like 2 settings:
- `max_splits` or `max_tokens`
- boolean to always return the remaining string in the last element

Also, `python` and `spark` is not clear at all unless user is familiar with it. 

What if it wants a behaviour of a different system but it doesn't know if it's python or spark mode?

Reply:
You are right.

I actually think it is sufficient to provide a single boolean setting which control if the remaining string is returned as last element or not. This was the main point of criticism in previous PRs.

The other setting would only control if N or N+1 elements are returned. It is easy for users to adjust N (to obtain the other "mode") but it is not easy to do that with the inclusion of the remaining string.

I'll also rename the setting name.